app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,38 @@
+class SessionsController < ApplicationController
+  skip_before_action :authenticate_request, only: %i[create logout]
+
+  def create
+    user = User
+      .find_by(email: params[:email])
+      .try(:authenticate, params[:password])
+    if user
+      token = jwt_encode(user_id: user.id)
+      render json: {
+        status: 'Logged in successfully!',
+        logged_in: true,
+        user:,
+        token:
+      }
+    else
+      render json: { status: 'Incorrect email or password.' }
+    end
+  end
+
+  def logged_in
+    if @current_user
+      render json: {
+        logged_in: true,
+        user: @current_user
+      }
+    else
+      render json: {
+        logged_in: false
+      }
+    end
+  end
+
+  def logout
+    reset_session
+    render json: { status: 200, logged_out: true }
+  end

```

Comment:
[OPTIONAL] Is it possible that some of this functionality could be handled on the model level? See the article referenced in the Main part of the review. (Of course, you don't have to change this, but it is something to consider)